cmd/kubelet/app/server.go

```diff
@@ -1296,3 +1304,35 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 	}
 	return rl, nil
 }
+
+func checkCRIStatus(runtimeService internalapi.RuntimeService, r *http.Request) error {
+	status, err := runtimeService.Status(false)
+	if err != nil {
+		return err
+	}
+	if _, found := r.URL.Query()["ignore_conditions"]; found {
+		return nil
+	}
+	requiredConditions := map[string]*runtimeapi.RuntimeCondition{
+		"RuntimeReady": nil,
+		"NetworkReady": nil,
```

Comment:
can we eliminate this and just keep to runtime?

we should not restart the kubelet if the cni is not yet deployed...

Reply:
Agree on eliminating NetworkReady but we should restart the container runtime (ie. containerd) rather kubelet for either case. WDYT?